documentation/sse_user_manual.md

```diff
@@ -0,0 +1,117 @@
+# CYBERCRYPT D1 Library - Symmetric Searchable Encryption user manual 
+
+This document explains how the Symmetric Searchable Encryption (SSE) implemented in the D1 Library works.
+
+## SSE scheme
+
+SSE allows users to search for keywords in encrypted data without decrypting it.
+
+The SSE scheme provides 3 algorithms, `Add`, `Search`, and `Delete`.
+
+(Usually, the elements in a hash map are called keys and values. In this context the keys are referred to as *labels* in order to not mix them up with encryption keys.) 
+
+* `Add` takes as input a keyword and an identifier (e.g. a document ID), computes an encrypted label -> value pair, and stores it in a secure index. The label -> value pair maps the given keyword to the given identifier, i.e. recognizes that the given keyword is contained in the given identifier. Every time a specific keyword is mapped to a new identifier, a slightly different version of the same label is used. That makes it possible to easily find all the identifiers that the keyword is mapped to, using a `Search` query.
+* `Search` takes as input a keyword, computes the corresponding encrypted labels, finds the encrypted values in the secure index that the labels map to, decrypts them, and returns the plaintext identifiers to the user.
+* `Delete` takes as input a keyword and an identifier, computes the encrypted label -> value pair, and deletes that pair the from the secure index. Note that `Delete` does not remove the keyword from the identifier, it only removes the keyword/identifier pair from the secure index.
```

Comment:
I'm not sure this makes sense. See my comment about SSE vs Secure Index. 
```suggestion
* `Delete` takes as input a keyword and an identifier, computes the encrypted label -> value pair, and deletes that pair the from the secure index. ```

Reply:
Fixed